ge/org.osate.ge/src/org/osate/ge/DiagramExporter.java

```diff
@@ -0,0 +1,296 @@
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.osate.ge.internal.diagram.runtime.AgeDiagram;
+import org.osate.ge.internal.diagram.runtime.DiagramSerialization;
+import org.osate.ge.internal.diagram.runtime.layout.DiagramElementLayoutUtil;
+import org.osate.ge.internal.diagram.runtime.layout.LayoutInfoProvider;
+import org.osate.ge.internal.graphiti.AgeDiagramTypeProvider;
+import org.osate.ge.internal.graphiti.AgeFeatureProvider;
+import org.osate.ge.internal.graphiti.diagram.ColoringProvider;
+import org.osate.ge.internal.graphiti.diagram.GraphitiAgeDiagram;
+import org.osate.ge.internal.services.ExtensionRegistryService;
+import org.osate.ge.internal.ui.editor.AgeDiagramBehavior;
+import org.osate.ge.internal.util.ProjectUtil;
+import org.osgi.framework.FrameworkUtil;
+
+public class DiagramExporter {
+	/**
+	 * This method receives an aadl diagram file and an output file.
+	 * The diagram contained in the aadl diagram file is written to
+	 * the output file as a png.
+	 *
+	 * @param  diagramFile the diagram file to be exported
+	 * @param  outputFile  the file the image will be written to
+	 */
+	public static void exportDiagramAsPng(final IFile diagramFile, final File outputFile) throws IOException {
+		// Image container bounds
+		final ContainerBounds containerBounds = new ContainerBounds();
+		final URI uri = URI.createPlatformResourceURI(diagramFile.getFullPath().toString(), true);
+		final IProject project = ProjectUtil.getProject(uri);
+
+		// Create diagram
+		final org.osate.ge.diagram.Diagram mmDiagram = DiagramSerialization.readMetaModelDiagram(uri);
+
+		final TransactionalEditingDomain editingDomain = TransactionalEditingDomain.Factory.INSTANCE
+				.createEditingDomain();
+		final DummyDiagramBehavior dummyDiagramBehavior = new DummyDiagramBehavior(editingDomain, project);
+		final DummyDiagramTypeProvider dummyDiagramTypeProvider = new DummyDiagramTypeProvider(dummyDiagramBehavior);
+
+		final AgeDiagram ageDiagram = DiagramSerialization.createAgeDiagram(project, mmDiagram,
+				getExtensionRegistryService(dummyDiagramTypeProvider.getClass()));
+
+		// Display warning if the diagram is stored with a newer version of the diagram file format.
+		if (mmDiagram.getFormatVersion() > DiagramSerialization.FORMAT_VERSION) {
+			MessageDialog.openWarning(Display.getCurrent().getActiveShell(),
+					"Diagram Created with Newer Version of OSATE", "The diagram '" + uri.lastSegment()
+					+ "' was created with a newer version of the OSATE. The diagram may not be correctly displayed. Saving the diagram with the running version of OSATE may result in the loss of diagram information.");
+		}
+
+		// Create an empty Graphiti diagram
+		final org.eclipse.graphiti.mm.pictograms.Diagram diagram = Graphiti.getPeService()
+				.createDiagram(GraphitiAgeDiagram.AADL_DIAGRAM_TYPE_ID, "", true);
+
+		final GraphitiAgeDiagram graphitiAgeDiagram = layoutAndCreateGraphitiAgeDiagram(dummyDiagramTypeProvider, ageDiagram,
```

Comment:
Use try with resourcest ensure the diagram is closed. A try.. finally.. is used later but unchecked exceptions could be thrown an other times.

See: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html